tests/Unit/Traits/TestTraits/PhpUnit/TestDatabaseProfilerTraitTest.php

```diff
@@ -26,16 +26,450 @@ public function testStopDatabaseQueryLog(): void
         $this->stopDatabaseQueryLog();
     }
 
-    protected function setUp(): void
+    public function testGetDatabaseQueries(): void
     {
-        parent::setUp();
-        $this->trait = new class {
-            use TestDatabaseProfilerTrait;
-        };
```

Comment:
@yalsicor Let me explain the idea behind why I did this.
I deliberately tested the trait this way to test it in isolation and independent of it's usage. It "just happens" that it is used in the parent class and is accessible in this case via its parent class (TestCase).
After testing it in isolation, then it should be enough to just test "if the trait is being used" in the TestCase or not.
If TestCase is using this trait, then it should work because we have a test for the trait.

Reply:
@Mohammad-Alavi I understand that, its merely that I cannot test the methods on `$this->trait`, because they are `protected`, which is in line with how methods are defined in `TestAssertionHelperTrait`. How would you test the methods there?

The only thing I can think of is make the methods public.

Reply:
@yalsicor I think this is an special case. Let's go with how it is now. Ill merge this later today if you are done with the PR 💪